core/include/gnuradio-4.0/Scheduler.hpp

```diff
@@ -17,24 +19,29 @@ using gr::thread_pool::BasicThreadPool;
 
 enum ExecutionPolicy { singleThreaded, multiThreaded };
 
-template<profiling::ProfilerLike TProfiler = profiling::null::Profiler>
+constexpr std::chrono::milliseconds kMessagePollInterval{ 10 };
+
+template<typename Derived, profiling::ProfilerLike TProfiler = profiling::null::Profiler>
 class SchedulerBase {
-public:
     lifecycle::State _state = lifecycle::State::IDLE;
 
 protected:
     gr::Graph                           _graph;
     TProfiler                           _profiler;
     decltype(_profiler.forThisThread()) _profiler_handler;
     std::shared_ptr<BasicThreadPool>    _pool;
-    std::atomic_uint64_t                _progress;
-    std::atomic_size_t                  _running_threads;
+    std::atomic_size_t                  _running_jobs;
     std::atomic_bool                    _stop_requested;
 
     MsgPortOutNamed<"__ForChildren"> _toChildMessagePort;
     MsgPortInNamed<"__FromChildren"> _fromChildMessagePort;
 
-    std::string unique_name = gr::meta::type_name<SchedulerBase<TProfiler>>(); // TODO: to be replaced if scheduler derives from Block<T>
+    std::string unique_name = gr::meta::type_name<Derived>(); // TODO: to be replaced if scheduler derives from Block<T>
+
+    [[nodiscard]] constexpr lifecycle::State
+    state() const {
```

Comment:
-> should become obsolete through using `lifecycle::StateMachine`.

Reply:
indeed